content/blog/2019/08/2019-08-17-folder-auth-plugin.adoc

```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: "Introducing Folder Auth Plugin v1.0"
+tags:
+- plugins
+- security
+- performance
+- gsoc
+- gsoc2019
+author: abhyudayasharma
+---
+
+During my link:/projects/gsoc/2019/role-strategy-performance[Google Summer of Code Project],
+I have created the brand new plugin:folder-auth[Folder Auth Plugin] for easily
+managing permissions to projects organized in folders from the plugin:cloudbees-folder[Folders plugin].
+This new plugin is designed for fast permission checks with easy-to-manage roles.
+The 1.0 version of the plugin has just been released and can be downloaded
+from your Jenkins' Update center.
+
+image::/images/post-images/gsoc-folder-auth-plugin/folder-auth.png[Screenshot of the Folder Auth Plugin]
+
+This plugin was inspired by the plugin:role-strategy[Role Strategy Plugin]
+and brings about performance improvements and makes managing roles much easier.
+The plugin supports three types of roles which are applicable at different places
+in Jenkins.
+
+* Global Roles: applicable everywhere in Jenkins
+* Agent Roles: restrict permissions for multiple agents connected to your instance
+* Folder Roles: applicable to multiple jobs organized inside folders
+
+This plugin, unlike the Role Strategy plugin, does not use regular expressions
+for finding matching projects and agents giving us performance improvements
+and makes administrators' lives easier. To reduce the number of roles required
+to be managed, permissions given to a folder through a folder role get inherited
+to all of its children. This is useful for giving access to multiple projects
+through a single role. Similarly, an agent role can be applied to multiple agents
+and assigned to multiple users.
+
+==== Performance Improvements over Role Strategy Plugin
+
+This plugin is designed to outperform Role Strategy Plugin in permission
+checks. The improvements were measured using the 
+link:blog/2019/06/21/performance-testing-jenkins/[micro-benchmark framework]
+I had created during the first phase of my GSoC project.
+Benchmarks for identical configurations for both plugin show that the
+permissions check are up to 934x faster for 500 global roles when compared to
+the global roles from the Role Strategy 2.13, which in itself contains several
+performance improvements. Comparing folder roles with Role Strategy's project
+roles, a permission check for access to a job almost 15x faster.
```

Comment:
Add some benchmarks here? Or maybe describe the test scenarios a bit?

Reply:
May use a table to compare